pylsp/plugins/rope_autoimport.py

```diff
@@ -238,3 +238,15 @@ def pylsp_document_did_open(config: Config, workspace: Workspace):
 def pylsp_document_did_save(config: Config, workspace: Workspace, document: Document):
     """Update the names associated with this document."""
     _reload_cache(config, workspace, [document])
+
+
+@hookimpl
+def pylsp_workspace_configuration_chaged(config: Config, workspace: Workspace):
+    """
+    Initialize autoimport if it has been enabled through a
+    workspace/didChangeConfiguration message from the frontend.
+
+    Generates the cache for local and global items.
+    """
+    if config.plugin_settings("rope_autoimport").get("enabled", False):
+        _reload_cache(config, workspace)
```

Comment:
It could keep the last enabled state and only reload cache on setting actually changing. Otherwise it can trigger wasteful reloads on any other setting changing.

Reply:
Yeah, that's a good very idea, thanks @rchl!

@tkrabel-db, what do you think? We could use a global variable to track the plugin's enabled state and only reload the cache when changing from `ENABLED = False` to `True`.

Reply:
Will do that!